plugins/module_utils/network/ios/config/acls/acls.py

```diff
@@ -101,21 +101,35 @@ def _compare(self, want, have, afi):
         for the acls network resource.
         """
 
+        def rearrange_cmds(aces):
+            non_negates = []
+            negates = []
+            for ace in aces:
+                if ace.startswith("no"):
+                    negates.append(ace)
+                else:
+                    non_negates.append(ace)
+            if non_negates or negates:
+                negates.extend(non_negates)
+            return negates
+
         wplists = want.get("acls", {})
         hplists = have.get("acls", {})
         for wname, wentry in iteritems(wplists):
             hentry = hplists.pop(wname, {})
             acl_type = wentry["acl_type"] if wentry.get("acl_type") else hentry.get("acl_type")
-            begin = len(
-                self.commands,
-            )  # to determine the index for acl command
+            begin = len(self.commands)  # to determine the index for acl command
             self._compare_aces(
                 wentry.pop("aces", {}),
                 hentry.pop("aces", {}),
                 afi,
                 wname,
             )  # handle aces
 
+            self.commands[begin : len(self.commands)] = rearrange_cmds(
+                self.commands[begin : len(self.commands)],
```

Comment:
I think this is the solution to the sanity issue.
```suggestion
            self.commands[begin:len(self.commands)] = rearrange_cmds(
                self.commands[begin:len(self.commands)],
```

Reply:
Ugh yeah, black is the one introducing this change. We will have to do something about that.

Reply:
Is `len(self.commands)` meant to change over the course of execution? Can we not instead do something like this?
```suggestion
            end = len(self.commands)
            self.commands[begin:end] = rearrange_cmds(
                self.commands[begin:],
```